### PR TITLE
HttpDateParse: Avoid unnecessary char[] allocation and cleanup

### PR DESCRIPTION
**Pull request by justinvp**
Avoid calling `ToCharArray` and just operate on the string directly to avoid the unnecessary `char[]` allocation. Also, cleaned the formatting and naming in the file while making changes here.

cc: @davidsh, @CIPop, @Priya91

### CODE REVIEW

**Comment by davidsh** [comment]
Where does this class get used in System.Net.Requests?  Do you think we have enough test coverage to make us confident of this change?

File: src/System.Net.Requests/src/System/Net/HttpDateParse.cs [hunk 1]
```diff
@@ -179,143 +146,109 @@ int index
...
+                {
                     //
                     // we have a string, should be a day, month, or GMT
                     //   lets skim to the end of the string
                     //
 
-                    rgdwDateParseResults[i] =
-                    MapDayMonthToDword(lpInputBuffer, index);
+                    dateParseResults[i] =
+                    MapDayMonthToDword(dateString, index);
```
Comment (stephentoub) [inline src/System.Net.Requests/src/System/Net/HttpDateParse.cs]: Nit: as long as you're fixing formatting, this could be on the same line.

**Comment by justinvp** [comment]
> Where does this class get used in System.Net.Requests?

It's used by `HttpWebRequest.IfModifiedSince`, `HttpWebRequest.Date`, and `HttpWebResponse.LastModified`.

> Do you think we have enough test coverage to make us confident of this change?

I didn't check the line coverage, but as a one-off test, I did change the implementation to always return `DateTime.Now` and ran the tests and several tests failed. The tests pass with the actual implementation before/after the changes.

File: src/System.Net.Requests/src/System/Net/HttpDateParse.cs (hunk 1 above)
Comment (justinvp) [inline src/System.Net.Requests/src/System/Net/HttpDateParse.cs]: I meant to change that. Will fix.

**Pull request merged**